### PR TITLE
WT-3161 Panic on a write error in logging.

### PR DESCRIPTION
**Pull request by sueloverso**
@agorrod and @ddanderson Here's a fix for the logging hang.  Please review.  I changed it so that if writing to the log file gets an error we panic.

### CODE REVIEW

**Comment by ddanderson** [comment]
@sueloverso, I ran it against the wt2909 test a few times and it passes. The code change LGTM, but I don't feel qualified to to make a deep judgement about how it affects logging.

**Comment by sueloverso** [comment]
@agorrod Give this a look for merging please.

File: src/log/log.c [hunk 1]
```diff
@@ -21,6 +21,27 @@ static int __log_write_internal(
 #define	WT_LOG_OPEN_VERIFY	0x02
 
 /*
+ * __log_panic --
+ *	Report an error, then panic the system.
+ */
+static int
```
Comment (agorrod) [inline src/log/log.c]: I believe the ``WT_PANIC_MSG`` macro mirrors the functionality you added here.

I'd be happy to change our ``__wt_panic`` to require an error message instead - I think that's a reasonable requirement for an operation that intends to panic the database.

I looked and the check for ``WT_CONN_PANIC`` on API entry doesn't result in another call to ``__wt_panic``, so changing that function shouldn't result in getting an error message from every thread.

Comment (sueloverso) [inline src/log/log.c]: I think `WT_PANIC_MSG` is the right thing here.  Fixed.

**Comment by agorrod** [comment]
Thanks @sueloverso lgtm

**Pull request merged**